components/brave_sync/brave_sync_service_impl.cc

```diff
@@ -0,0 +1,626 @@
+
+  std::unique_ptr<jslib::Device> device_record =
+      std::make_unique<jslib::Device>();
+  device_record->name = device->name_;
+  record->SetDevice(std::move(device_record));
+
+  return record;
+}
+
+}  // namespace
+
+BraveSyncServiceImpl::BraveSyncServiceImpl(Profile* profile) :
+    sync_client_(BraveSyncClient::Create(this, profile)),
+    sync_initialized_(false),
+    sync_words_(std::string()),
+    profile_(profile),
+    sync_prefs_(new brave_sync::prefs::Prefs(profile->GetPrefs())),
+    bookmark_change_processor_(BookmarkChangeProcessor::Create(
+        profile,
+        sync_client_.get(),
+        sync_prefs_.get())),
+    timer_(std::make_unique<base::RepeatingTimer>()),
+    unsynced_send_interval_(base::TimeDelta::FromMinutes(10)) {
+
+  // Moniter syncs prefs required in GetSettingsAndDevices
+  profile_pref_change_registrar_.Init(profile->GetPrefs());
+  profile_pref_change_registrar_.Add(
+      prefs::kSyncEnabled,
+      base::Bind(&BraveSyncServiceImpl::OnSyncPrefsChanged,
+                 base::Unretained(this)));
+  profile_pref_change_registrar_.Add(
+      prefs::kSyncDeviceName,
+      base::Bind(&BraveSyncServiceImpl::OnSyncPrefsChanged,
+                 base::Unretained(this)));
+  profile_pref_change_registrar_.Add(
+      prefs::kSyncDeviceList,
+      base::Bind(&BraveSyncServiceImpl::OnSyncPrefsChanged,
+                 base::Unretained(this)));
+  profile_pref_change_registrar_.Add(
+      prefs::kSyncBookmarksEnabled,
+      base::Bind(&BraveSyncServiceImpl::OnSyncPrefsChanged,
+                 base::Unretained(this)));
+  profile_pref_change_registrar_.Add(
+      prefs::kSyncSiteSettingsEnabled,
+      base::Bind(&BraveSyncServiceImpl::OnSyncPrefsChanged,
+                 base::Unretained(this)));
+  profile_pref_change_registrar_.Add(
+      prefs::kSyncHistoryEnabled,
+      base::Bind(&BraveSyncServiceImpl::OnSyncPrefsChanged,
+                 base::Unretained(this)));
+
+  if (!sync_prefs_->GetSeed().empty() &&
+      !sync_prefs_->GetThisDeviceName().empty()) {
+    sync_configured_ = true;
+  }
+}
+
+BraveSyncServiceImpl::~BraveSyncServiceImpl() {}
+
+BraveSyncClient* BraveSyncServiceImpl::GetSyncClient() {
+  return sync_client_.get();
+}
+
+bool BraveSyncServiceImpl::IsSyncConfigured() {
+  return sync_configured_;
+}
+
+bool BraveSyncServiceImpl::IsSyncInitialized() {
+  return sync_initialized_;
+}
+
+void BraveSyncServiceImpl::Shutdown() {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+  bookmark_change_processor_->Stop();
+
+  StopLoop();
+}
+
+void BraveSyncServiceImpl::OnSetupSyncHaveCode(const std::string& sync_words,
+    const std::string& device_name) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+  if (sync_words.empty() || device_name.empty()) {
+    OnSyncSetupError("missing sync words or device name");
+    return;
+  }
+
+  if (initializing_) {
+    NotifyLogMessage("currently initializing");
```

Comment:
I'm not sure I understand `NotifyLogMessage`. Why are we sending this to the background page instead of just logging normally in the console?

Reply:
let's look at this in a follow-up because I don't know that we need this method

Reply:
it is indeed does nothing except just logging. So I removed pass into js UI layer.